### DISCUSSION TITLE
Are `nerdctl` managed containers expected to show up in `crictl` output?

### DISCUSSION THREAD

**Discussion by greyhame-s** [containerd/nerdctl#1676] [Q&A] [answered]
I see the FAQs section that mentions that `nerdctl` [doesn't use the CRI API](https://github.com/containerd/nerdctl/blob/main/docs/faq.md#nerdctl-ignores-pluginsiocontainerdgrpcv1cri-config). Given that, is it expected that `crictl` wouldn't report containers created via `nerdctl` since it uses the CRI gRPC calls?

Both the clients were pointed to the same socket `/var/run/containerd/containerd.sock`, and were being run as the superuser.

**Comment by greyhame-s** [comment]
I should have mentioned that the test environment is a stand alone container, and not part of a cluster.

**Comment by AkihiroSuda** [comment, marked as answer]
nerdctl containers don't show up in crictl ~unless they were launched with `nerdctl --namespace=k8s.io`~

**Reply by greyhame-s** [reply]
Thanks for the prompt response! I specifically tried to create the nerdctl container in that specific namespace and it still doesn't show up in `crictl` output. Is it possible to have the CRI plugin misconfigured in order to get into this error scenario? As expected `ctr` reports the correct container.

```
nerdctl --namespace k8s.io run -it --rm --name container-name ...
```

```
sudo crictl -r unix:///run/containerd/containerd.sock ps -a
CONTAINER           IMAGE               CREATED             STATE               NAME                ATTEMPT             POD ID              POD

crictl  -r unix:///run/containerd/containerd.sock ps -a
CONTAINER           IMAGE               CREATED             STATE               NAME                ATTEMPT             POD ID              POD

```

**Reply by AkihiroSuda** [reply]
Sorry, I forgot to mention that crictl only show CRI containers.
nerdctl containers can never be shown in crictl.
